index.d.ts

```diff
@@ -0,0 +1,1100 @@
+// Type definitions for [~THE LIBRARY NAME~] [~OPTIONAL VERSION NUMBER~]
+// Project: [~THE PROJECT NAME~]
+// Definitions by: [~YOUR NAME~] <[~A URL FOR YOU~]>
+
+export as namespace Draggable;
+
+// type definitions
+export type DraggableContainer = HTMLElement[] | NodeList | HTMLElement
+
+type DraggableEventName = 'draggable:initialized' | 'draggable:destroy' | string
+type DragEventName = 'drag:start' | 'drag:move' | 'drag:over' | 'drag:over:container' | 'drag:out' | 'drag:out:container' | 'drag:stop' | 'drag:pressure' | string
+type SortEventName = 'sortable:start' | 'sortable:sorted' | 'sortable:stop' | string
+type MirrorEventName = 'mirror:created' | 'mirror:attached' | 'mirror:move' | 'mirror:destroy' | string
+
+// plugin
+export class DraggablePlugin {
```

Comment:
There is no actual newable for plugins iirc (@tsov am I right?).

What we actually want to do here is export a type that is `typeof DraggablePlugin` so that the interface for a valid plugin class is usable, but there are no erroneous nonexistant class functions.

Reply:
maybe the library should also export an base valid plugin class for users to extend, same as sensor

Reply:
Yes, I think it makes sense to provide a base Plugin class to extend from. Will add this to the `beta.4` release 👍 What's the best thing we can do for now?

Reply:
imo `typeof` for this type def so nobody is misled, and then just update it when we have a base class.